stackslib/src/net/api/postblock_v3.rs

```diff
@@ -0,0 +1,185 @@
+                "Invalid Http request: PostBlock body is too big".to_string(),
+            ));
+        }
+
+        if Some(HttpContentType::Bytes) != preamble.content_type || preamble.content_type.is_none()
+        {
+            return Err(Error::DecodeError(
+                "Invalid Http request: PostBlock takes application/octet-stream".to_string(),
+            ));
+        }
+
+        let block = Self::parse_postblock_octets(body)?;
+
+        self.block = Some(block);
+        Ok(HttpRequestContents::new().query_string(query))
+    }
+}
+
+impl RPCRequestHandler for RPCPostBlockRequestHandler {
+    /// Reset internal state
+    fn restart(&mut self) {
+        self.block = None;
+    }
+
+    /// Make the response
+    fn try_handle_request(
+        &mut self,
+        preamble: HttpRequestPreamble,
+        _contents: HttpRequestContents,
+        node: &mut StacksNodeState,
+    ) -> Result<(HttpResponsePreamble, HttpResponseContents), NetError> {
+        // get out the request body
+        let block = self
+            .block
+            .take()
+            .ok_or(NetError::SendError("`block` not set".into()))?;
+
+        let response = node
+            .with_node_state(|network, sortdb, chainstate, _mempool, _rpc_args| {
+                let mut handle_conn = sortdb.index_handle_at_tip();
+                Relayer::process_new_nakamoto_block(
+                    &network.burnchain,
+                    &sortdb,
+                    &mut handle_conn,
+                    chainstate,
+                    &block,
+                    None,
+                )
+            })
+            .map_err(|e| {
+                StacksHttpResponse::new_error(&preamble, &HttpError::new(400, e.to_string()))
+            });
+
+        let data_resp = match response {
+            Ok(accepted) => StacksBlockAcceptedData {
+                accepted,
+                stacks_block_id: block.block_id(),
+            },
+            Err(e) => {
+                return e.try_into_contents().map_err(NetError::from);
+            }
+        };
+
+        // should set to relay...
+        if data_resp.accepted {
+            node.set_relay_message(StacksMessageType::NakamotoBlocks(NakamotoBlocksData {
+                blocks: vec![block],
+            }));
+        }
+
+        let mut preamble = HttpResponsePreamble::ok_json(&preamble);
+        preamble.set_canonical_stacks_tip_height(Some(node.canonical_stacks_tip_height()));
+        let body = HttpResponseContents::try_from_json(&data_resp)?;
+        Ok((preamble, body))
+    }
+}
+
+/// Decode the HTTP response
+impl HttpResponse for RPCPostBlockRequestHandler {
+    fn try_parse_response(
+        &self,
+        preamble: &HttpResponsePreamble,
+        body: &[u8],
+    ) -> Result<HttpResponsePayload, Error> {
+        let accepted: StacksBlockAcceptedData = parse_json(preamble, body)?;
+        Ok(HttpResponsePayload::try_from_json(accepted)?)
```

Comment:
Could be simplified to

```suggestion
        HttpResponsePayload::try_from_json(accepted)
```